core/arch/arm/kernel/ree_fs_ta.c

```diff
@@ -112,6 +118,70 @@ static TEE_Result rpc_load(const TEE_UUID *uuid, struct shdr **ta,
 	return res;
 }
 
+static TEE_Result check_update_version(struct shdr_bootstrap_ta *hdr)
```

Comment:
Perhaps we want to make this configurable via `CFG_REE_FS_TA_ROLLBACK_PROTECTION ?= y`? And warn or even error out if `CFG_RPMB_FS != y` because in this case the extra protection is an illusion.

Reply:
`CFG_RPMB_FS` is disabled by default, so I am not sure if we would like to get even warning on default builds.

> in this case the extra protection is an illusion.

This illusion is true for Secure Storage TAs as well and perhaps secure storage used by any TAs that expects it to be non-modifiable via any unauthorised entity, but with rollback allowed the purpose is somewhat defeated.

So either we rely on REE for rollback protection as stated in this [GP requirement](https://optee.readthedocs.io/en/latest/architecture/secure_storage.html#globalplatform-trusted-storage-requirement) or we see it as test system with `CFG_RPMB_FS=n`.

Reply:
OK, fair enough.